pkg/runtime/runtime_test.go

```diff
@@ -5392,3 +5371,201 @@ func TestNamespacedPublisher(t *testing.T) {
+		"testInputBinding",
+	)
+	cin := componentsV1alpha1.Component{}
+	cin.ObjectMeta.Name = "testInputBinding"
+	cin.Spec.Type = "bindings.testInputBinding"
+
+	rt.bindingsRegistry.RegisterOutputBinding(
+		func(_ logger.Logger) bindings.OutputBinding {
+			return &daprt.MockBinding{}
+		},
+		"testOutputBinding",
+	)
+	cout := componentsV1alpha1.Component{}
+	cout.ObjectMeta.Name = "testOutputBinding"
+	cout.Spec.Type = "bindings.testOutputBinding"
+	require.NoError(t, rt.initInputBinding(cin))
+	require.NoError(t, rt.initOutputBinding(cout))
+
+	assert.Equal(t, len(rt.inputBindings), 1)
+	assert.Equal(t, len(rt.outputBindings), 1)
+
+	rt.runtimeConfig.GracefulShutdownDuration = 3 * time.Second
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sigs
+		rt.Shutdown(rt.runtimeConfig.GracefulShutdownDuration)
+	}()
+	if p, err := os.FindProcess(os.Getpid()); err != nil {
+		rt.Shutdown(rt.runtimeConfig.GracefulShutdownDuration)
+	} else {
+		p.Signal(syscall.SIGTERM)
+	}
+	time.Sleep(1 * time.Second)
+	assert.Nil(t, rt.inputBindingsCancel)
+	assert.Nil(t, rt.inputBindingsCtx)
+}
+
+func TestGracefulShutdownActors(t *testing.T) {
+	rt := NewTestDaprRuntime(modes.StandaloneMode)
+
+	bytes := make([]byte, 32)
+	rand.Read(bytes)
+	primaryKey := hex.EncodeToString(bytes)
+
+	mockStateComponent := componentsV1alpha1.Component{
+		ObjectMeta: metaV1.ObjectMeta{
+			Name: TestPubsubName,
+		},
+		Spec: componentsV1alpha1.ComponentSpec{
+			Type:    "state.mockState",
+			Version: "v1",
+			Metadata: []componentsV1alpha1.MetadataItem{
+				{
+					Name: "actorStateStore",
+					Value: componentsV1alpha1.DynamicValue{
+						JSON: v1.JSON{Raw: []byte("true")},
+					},
+				},
+				{
+					Name: "primaryEncryptionKey",
+					Value: componentsV1alpha1.DynamicValue{
+						JSON: v1.JSON{Raw: []byte(primaryKey)},
+					},
+				},
+			},
+		},
+		Auth: componentsV1alpha1.Auth{
+			SecretStore: "mockSecretStore",
+		},
+	}
+
+	// setup
+	initMockStateStoreForRuntime(rt, primaryKey, nil)
+
+	// act
+	err := rt.initState(mockStateComponent)
+
+	// assert
+	assert.NoError(t, err, "expected no error")
+
+	rt.namespace = "test"
+	rt.runtimeConfig.mtlsEnabled = true
+	assert.Nil(t, rt.initActors())
+
+	rt.runtimeConfig.GracefulShutdownDuration = 3 * time.Second
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sigs
+		rt.Shutdown(rt.runtimeConfig.GracefulShutdownDuration)
+	}()
+
+	if p, err := os.FindProcess(os.Getpid()); err != nil {
+		rt.Shutdown(rt.runtimeConfig.GracefulShutdownDuration)
+	} else {
+		p.Signal(syscall.SIGTERM)
+	}
+	time.Sleep(3 * time.Second)
+
+	var activeActCount int
+	activeActors := rt.actor.GetActiveActorsCount(rt.ctx)
+	for _, v := range activeActors {
+		activeActCount += v.Count
+	}
+	assert.Equal(t, activeActCount, 0)
+}
+
+func initMockStateStoreForRuntime(rt *DaprRuntime, primaryKey string, e error) *daprt.MockStateStore {
```

Comment:
```suggestion
func initMockStateStoreForRuntime(rt *DaprRuntime, encKey string, e error) *daprt.MockStateStore {
```
to not get confused with `primaryKey` in a state store ... Please change all occurrences while calling also if possible.

Reply:
Updated. Please check @mukundansundar